tests/commerce_integration/TEC/Tickets/OrphanedPostsTest.php

```diff
@@ -0,0 +1,320 @@
+
+		// Delete the event.
+		wp_delete_post( $event_id, true );
+
+		// First call to populate cache.
+		$orphaned_posts_1 = tribe( $provider )->get_orphaned_post_ids( $provider );
+
+		// Get cached results.
+		$cache_key         = 'tec_tickets_orphaned_posts_' . sanitize_key( $provider );
+		$cached_post_ids_1 = get_transient( $cache_key );
+
+		// Results should be identical (cached).
+		$this->assertEquals( $orphaned_posts_1, $cached_post_ids_1 );
+
+		// Delete one of the orphaned posts to trigger cache invalidation.
+		wp_delete_post( $rsvp_ticket_id, true );
+
+		// Cache should be invalid (false).
+		$cached_post_ids_2 = get_transient( $cache_key );
+		$this->assertFalse( $cached_post_ids_2 );
+
+		// Get the updated orphaned posts.
+		$orphaned_posts_2 = tribe( $provider )->get_orphaned_post_ids( $provider );
+
+		$post_ids = [ $rsvp_ticket_id, ...$rsvp_attendee_ids ];
+
+		// The deleted ticket should be in the first result (before deletion).
+		$this->assertEquals( $post_ids, $orphaned_posts_1 );
+
+		// The deleted ticket should not be in the second result, only the attendees should be.
+		$this->assertEquals( $rsvp_attendee_ids, $orphaned_posts_2 );
+	}
+
+	/**
+	 * Test with no orphaned posts.
+	 *
+	 * @test
+	 */
+	public function should_return_empty_array_when_no_orphaned_posts() {
+		// Get orphaned post IDs when there are no orphaned posts.
+		$orphaned_posts = tribe( 'Tribe__Tickets__RSVP' )->get_orphaned_post_ids( 'Tribe__Tickets__RSVP' );
+
+		$this->assertEmpty( $orphaned_posts );
+		$this->assertIsArray( $orphaned_posts );
+	}
+
+	/**
+	 * Test mixed providers scenario.
+	 *
+	 * @test
+	 */
+	public function should_handle_mixed_providers_correctly() {
+		// Create an event that will be deleted.
+		$event_id = tribe_events()->set_args( [
+			'title'      => 'Test Event for Mixed Providers',
+			'status'     => 'publish',
+			'start_date' => '2025-01-01 00:00:00',
+			'duration'   => 2 * HOUR_IN_SECONDS,
+		] )->create()->ID;
+
+		// Create both RSVP and TC tickets.
+		$rsvp_ticket_id = $this->create_rsvp_ticket( $event_id );
+		$tc_ticket_id = $this->create_tc_ticket( $event_id );
+
+		// Create orders and attendees for both ticket types.
+		$rsvp_attendee_ids = $this->create_many_attendees_for_ticket( 2, $rsvp_ticket_id, $event_id );
+		$tc_order_id = $this->create_order( [ $tc_ticket_id => 3 ] );
+
+		// Get attendees from the order.
+		$tc_attendee_ids = tribe( Module::class )->get_attendees_by_order_id( $tc_order_id );
+		$tc_attendee_ids = wp_list_pluck( $tc_attendee_ids, 'ID' );
+
+		// Delete the event.
+		wp_delete_post( $event_id, true );
+
+		// Get orphaned posts for RSVP provider.
+		$rsvp_orphaned = tribe( 'Tribe__Tickets__RSVP' )->get_orphaned_post_ids( 'Tribe__Tickets__RSVP' );
+
+		// Get orphaned posts for TC provider.
+		$tc_orphaned = tribe( Module::class )->get_orphaned_post_ids( 'TEC\Tickets\Commerce\Module' );
```

Comment:
That API is ... not ideal.

What should be enough is this:

```php
tribe( Module::class )->get_orphaned_post_ids();
tribe( 'Tribe__Tickets__RSVP' )->get_orphaned_post_ids();
```

Both of those should already found the appropriate orhaned tickets for their provider.

Reply:
Right. Fixed all across the test.